rest/admin_api.go

```diff
@@ -40,6 +40,11 @@ func (h *handler) handleCreateDB() error {
 	}
 	config.Name = dbName
 
+	bucket := dbName
+	if config.Bucket != nil {
+		bucket = *config.Bucket
+	}
+
```

Comment:
Whats the reason for moving this? 
It also seems a little misleading as we're not necessarily connecting to the `bucket` var in the non-persistent case